src/wrappers.jl

```diff
@@ -0,0 +1,442 @@
+        z(g1) == z(g2) || return false
+    end
+    if ismeasured(g1)
+        m(g1) == m(g2) || return false
+    end
+    return true
+end
+Base.:(!=)(g1::Point, g2::Point) = !(g1 == g2)
+
+@noinline _coord_length_error(Z, M, l) =
+    throw(ArgumentError("Number of coordinates must be $(2 + Z + M) when `Z` is $Z and `M` is $M. Got $l"))
+@noinline _no_z_error() = throw(ArgumentError("Point has no `Z` coordinate"))
+@noinline _no_m_error() = throw(ArgumentError("Point has no `M` coordinate"))
+
+"""
+    Feature(geometry; [properties, crs, extent])
+
+A Feature wrapper.
+
+## Arguments
+
+- `geometry`: any GeoInterface compatible geometry object, or `nothing`.
+
+## Keywords
+
+- `properties`: any object that defines `propertynames` and `getproperty`
+- `crs`: Any GeoFormatTypes.jl crs type, or `nothing`
+- `extent`: An Extents.jl `Extent` or `nothing`
+
+## Example
+
+```julia
+feature = Feature(geom; properties=(a=1, b="2"), crs=EPSG(4326))
+```
+"""
+struct Feature{T,C,E<:Union{Extents.Extent,Nothing}}
+    parent::T
+    crs::C
+    extent::E
+end
+function Feature(f::Feature; crs=f.crs, extent=f.extent, properties=nothing)
+    isnothing(properties) || @info "`properties` keyword not used when wrapping a feature"
+    Feature(parent(f), crs, extent)
+end
+function Feature(geometry=nothing; properties=nothing, crs=nothing, extent=nothing)
+    if isnothing(geometry) || isgeometry(geometry)
+        # Wrap a NamedTuple feature
+        Feature((; geometry, properties...), crs, extent)
+    else
+        throw(ArgumentError("object must be a feature, geometry or `nothing`. Got $(typeof(geometry))"))
+    end
+end
+
+Base.parent(f::Feature) = f.parent
+
+isfeature(::Type{<:Feature}) = true
+trait(::Feature) = FeatureTrait()
+geometry(f::Feature) = geometry(parent(f))
+properties(f::Feature) = properties(parent(f))
+extent(f::Feature) =
+    isfeature(parent(f)) && isnothing(f.extent) ? extent(parent(f)) : f.extent
+crs(f::Feature) =
+    isfeature(parent(f)) && isnothing(f.crs) ? crs(parent(f)) : f.crs
+
+"""
+    FeatureCollection(features; [crs, extent])
+
+A FeatureCollection wrapper.
+
+## Arguments
+
+- `features`: an `AbstractArray` of GeoInterface compatible features.
+    Iterables are accepted but will be collected to an `Array`.
+
+## Keywords
+
+- `crs`: Any GeoFormatTypes.jl crs type, or `nothing`
+- `extent`: An Extents.jl `Extent` or `nothing`
+
+## Examples
+
+```julia
+fc = FeatureCollection([feature1, feature2, feature3];
+    crs=EPSG(4326), extent=Extent(X=(11.0, 34.0), Y=(45.7, 78.0))
+)
+```
+
+Or wrap another `FeatureColection`, e.g. if it has no crs attached:
+
+```julia
+fc = FeatureCollection(featurecollection, crs=EPSG(4326))
+```
+"""
+struct FeatureCollection{P,C,E}
+    parent::P
+    crs::C
+    extent::E
+end
+function FeatureCollection(fc::FeatureCollection; crs=crs(fc), extent=extent(fc))
+    FeatureCollection(parent(fc), crs, extent)
+end
+function FeatureCollection(parent; crs=nothing, extent=nothing)
```

Comment:
One can override the crs and extent of a `parent` here, but those overrides are never returned anymore if `isfeaturecollection(parent)`.

Reply:
Good point, yes overrides should work in all cases